scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,683 @@
+import * as React from 'react';
+// @ts-ignore
+import ReactCrop from 'react-image-crop';
+import 'react-image-crop/dist/ReactCrop.css';
+import {ISuperdesk, IArticle} from 'superdesk-api';
+import {isEmpty, omit, pick, isEqual, cloneDeep, flatten} from 'lodash';
+
+import {VideoEditorTools} from './VideoEditorTools';
+import {VideoTimeline} from './VideoTimeline/VideoTimeline';
+import {VideoEditorHeader} from './VideoEditorHeader';
+import {VideoEditorThumbnail} from './VideoEditorThumbnail';
+import {IVideoEditor, IThumbnail} from './interfaces';
+
+interface IProps {
+    article: IArticle;
+    superdesk: ISuperdesk;
+    onClose: () => void;
+    onArticleUpdate: (articleUpdate: IArticle) => void;
+}
+
+interface IState extends IVideoEditor {
+    isDirty: boolean;
+    trim: {
+        start: number;
+        end: number;
+    };
+    cropImg: string;
+    thumbnails: Array<IThumbnail>;
+    loading: boolean;
+    loadingText: string;
+    scale: number;
+    videoSrc: string;
+    article: IArticle;
+}
+
+export class VideoEditor extends React.Component<IProps, IState> {
+    private videoRef: React.RefObject<HTMLVideoElement>;
+    private reactCropRef: React.RefObject<ReactCrop>;
+    private reactCropWrapperRef: React.RefObject<HTMLDivElement>;
+    private intervalThumbnails: number;
+    private intervalVideoEdit: number;
+    private intervalCheckVideo: number;
+    private initState: Pick<IState, 'crop' | 'degree' | 'trim' | 'quality'>;
+    private reactCropMarginDelta: number;
+    private hasTransitionRun: boolean;
+    wrapperSize: { width: number; height: number };
+
+    constructor(props: IProps) {
+        super(props);
+        this.videoRef = React.createRef();
+        this.reactCropRef = React.createRef();
+        this.reactCropWrapperRef = React.createRef();
+        this.intervalThumbnails = 0;
+        this.intervalVideoEdit = 0;
+        this.intervalCheckVideo = 0;
+        this.initState = {
+            crop: {aspect: 16 / 9, unit: 'px', scale: 1, width: 0, height: 0, x: 0, y: 0, value: 0},
+            degree: 0,
+            trim: {
+                start: 0,
+                end: 0,
+            },
+            quality: 0,
+        };
+        this.state = {
+            ...this.initState,
+            isDirty: false,
+            cropEnabled: false,
+            cropImg: '',
+            playing: false,
+            loading: false,
+            loadingText: '',
+            scale: 1,
+            thumbnails: [],
+            videoSrc: '',
+            article: cloneDeep(this.props.article),
+        };
+        this.wrapperSize = {width: 0, height: 0};
+        this.reactCropMarginDelta = 0;
+        this.hasTransitionRun = true;
+    }
+
+    componentDidMount() {
+        const canvas = document.createElement('canvas');
+
+        canvas.width = 2000;
```

Comment:
Whichever value is fine, as long as it's big enough to cover entire video. This is maximum area of React Crop, we will resize it to match video later, but it can't exceed init image size.
We will add a small note to it to make it more clear.